src/gpb_compile.erl

```diff
@@ -8489,7 +8506,7 @@ type_default(Type, Defs, Opts, GetTypeDefault) ->
     if Type == string ->
             case get_strings_as_binaries_by_opts(Opts) of
                 true ->
-                    list_to_binary(GetTypeDefault(Type, Defs));
+                    iolist_to_binary(GetTypeDefault(Type, Defs));
```

Comment:
Minor: Why change from `list_to_binary` into `iolist_to_binary`?
The default is never a binary here.
A reminiscence from the previous `strings_as_binaries` issue?